src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const cleanedSourceString = sourceString.replace(/\s*;\s*/g, ';').trim();
+  const strArray = cleanedSourceString.split(';');
+  const result = {};
+
+  strArray.forEach((str) => {
+    const [key, value] = str.split(':').map((s) => s.trim());
+
+    result[key] = value;
```

Comment:
This line will add `undefined` as a value if the `split` on line 18 does not find a colon in the string. You should guard against this to ensure no `undefined` values are added to the object.